src/data_storage/import/import_storage.ts

```diff
@@ -33,7 +35,17 @@ export default class ImportStorage extends PostgresStorage {
         return super.retrieve<ImportT>(ImportStorage.retrieveLastStatement(dataSourceID))
     }
 
-    public SetStatus(importID: string, status: "ready" | "processing" | "error" | "stopped" | "completed", message?: string): Promise<Result<boolean>> {
+    public async SetStatus(importID: string, status: "ready" | "processing" | "error" | "stopped" | "completed", message?: string): Promise<Result<boolean>> {
+        if (status === "completed") {
```

Comment:
Why not send an event regardless of the status itself? It would allow someone to listen to errors as well as success.

Reply:
Let's discuss this when you have time so I can make sure I understand the lifecycle of the status updates.

Reply:
Send on error/stopped/completed